packages/react-storage/src/components/StorageBrowser/Views/Controls/Table.tsx

```diff
@@ -441,6 +441,15 @@ export const LocationDetailViewTable = (): JSX.Element | null => {
               case 'download' as keyof LocationItem: {
```

Comment:
non-blocking, but does this still need a cast if the ones below don't?

Reply:
I think so, I think if we don't want to cast it as `LocationItem`, then we'll have to add it to the `FileItem` type

```
export interface FileItem {
  key: string;
  data?: File;
  lastModified: Date;
  size: number;
  type: 'FILE';
}
```

Reply:
Casting inside the input of a public callback as a code smell indicating potential friction points between the interface and its consumers that requires addressing.

In the context of React, a function that returns JSX  (a `render` function), providing `props` is idiomatic allowing for seamless passing of components as the callback parameter. 

Let's sync offline to track down where we need to modify our approach to address